### PR TITLE
Loading necessary modules to run this script on PowerVM

### PR DESCRIPTION
**Pull request by vrbagalkote**
Checking if the machine is booted with PowerVM hyp, if yes
loading the rpadlpar_io rpaphp modules, if not loaded already.

Signed-off-by: Venkat R B <vrbagal1@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vrbagalkote** [comment]
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1003441/job.txt)

File: io/pci/pci_hotplug.py [hunk 1]
```diff
@@ -43,14 +43,23 @@ def setUp(self):
         cmd = "uname -p"
         if 'ppc' not in process.system_output(cmd, ignore_status=True):
             self.skip("Processor is not ppc64")
+        if cpu._list_matches(cpu._get_cpu_info(), 'pSeries'):
+            for _ in ['rpaphp', 'rpadlpar_io']:
+                if not linux_modules.module_is_loaded("rpaphp"):
+                    linux_modules.load_module("rpaphp")
+                if not linux_modules.module_is_loaded("rpadlpar_io"):
```
Comment (narasimhan-v) [inline io/pci/pci_hotplug.py]:
```
for mdl in ['rpaphp', 'rpadlpar_io']:
    if not linux_modules.module_is_loaded(mdl):
        linux_modules.load_module(mdl)
```

File: io/pci/pci_hotplug.py [hunk 2]
```diff
@@ -43,14 +43,23 @@ def setUp(self):
...
+                linux_modules.load_module("php_pnv")
         self.return_code = 0
         self.device = self.params.get('pci_device', default=' ')
         if not os.path.isdir('/sys/bus/pci/devices/%s' % self.device):
             self.skip("PCI device given does not exist")
-        cmd = "cat /sys/bus/pci/devices/%s/devspec ; echo" % self.device
-        self.devspec = process.system_output(cmd, shell=True).strip('\n')
-        cmd = "cat /proc/device-tree/%s/ibm,loc-code ; echo" % self.devspec
-        self.slot = process.system_output(cmd, shell=True).strip('\0\n')
+        self.devspec = genio.read_file("/sys/bus/pci/devices/%s/devspec"
```
Comment (narasimhan-v) [inline io/pci/pci_hotplug.py]: :+1:

File: io/pci/pci_hotplug.py (hunk 1 above)
Comment (vrbagalkote) [inline io/pci/pci_hotplug.py]: oh yes,

File: io/pci/pci_hotplug.py [hunk 3]
```diff
@@ -43,14 +43,21 @@ def setUp(self):
...
+                linux_modules.load_module("pnv_php")
         self.return_code = 0
         self.device = self.params.get('pci_device', default=' ')
         if not os.path.isdir('/sys/bus/pci/devices/%s' % self.device):
             self.skip("PCI device given does not exist")
-        cmd = "cat /sys/bus/pci/devices/%s/devspec ; echo" % self.device
-        self.devspec = process.system_output(cmd, shell=True).strip('\n')
-        cmd = "cat /proc/device-tree/%s/ibm,loc-code ; echo" % self.devspec
-        self.slot = process.system_output(cmd, shell=True).strip('\0\n')
+        self.devspec = genio.read_file("/sys/bus/pci/devices/%s/devspec"
```
Comment (PraveenPenguin) [inline io/pci/pci_hotplug.py]: can  `self.devspec ` local variable as it seems me used out side `setUp` please consider

**Comment by vrbagalkote** [comment]
@PraveenPenguin , I have changed self.devspec to devspec

**Comment by PraveenPenguin** [comment]
@narasimhan-v thanks

**Pull request merged**